src/runtime/task/mod.rs

```diff
@@ -38,6 +41,74 @@ use waker::make_waker;
 
 pub(crate) const DEFAULT_INLINE_TASKS: usize = 16;
 
+/// To make debugging easier, if a task is assigned a `TaskName(s)` Label,
+/// Shuttle will display the String `s` in addition to the `TaskId` in debug output.
+#[derive(Clone, PartialEq, Eq)]
+pub struct TaskName(String);
+
+impl From<String> for TaskName {
+    fn from(s: String) -> Self {
+        Self(s)
+    }
+}
+
+impl From<&str> for TaskName {
+    fn from(s: &str) -> Self {
+        Self(String::from(s))
+    }
+}
+
+impl std::fmt::Debug for TaskName {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<TaskName> for String {
+    fn from(task_name: TaskName) -> Self {
+        task_name.0
+    }
+}
+
+impl<'a> From<&'a TaskName> for &'a String {
+    fn from(task_name: &'a TaskName) -> Self {
+        &task_name.0
+    }
+}
+
+/// By default, when a task or thread T is spawned, it inherits all labels from its parent.
```

Comment:
nit: The docs style is to write a one-sentence summary (which shows up on index pages) and then the rest of the text as a second+ paragraph.